plugins/snowflake/dbt/include/snowflake/macros/materializations/table.sql

```diff
@@ -35,10 +35,14 @@
   {{ adapter.drop_relation(intermediate_relation) }}
 
   -- setup: if the target relation already exists, truncate or drop it (if it's a view)
+  {# TODO: Would like to check this. New materialsiation makes these tests a bit moot. We should
```

Comment:
Let's support `--non-destructive` for now, but we might cruise through the materializations and rip it out for 0.14.0. It's not hurting anyone at the moment, but removing it will certainly simplify this code! For now though, let's continue to support it on Snowflake

Reply:
Ok sure makes sense to keep things consistant. So you would like be to add the truncate back etc? Because right now it will be doing pretty much nothing. Looking at this block again I see the following:
L#30-34, dropping the view is pretty much taken care of later between lines 46-49. 
Adding back this part of the code:
```  {% if non_destructive_mode -%}
    {% if exists_as_table -%}
      --noop we can do away with this step all together since the table can be replaced in Snowflake.
      {{ adapter.truncate_relation(old_relation) }}
```
Would cause the table to be trucated, which I think is more "destructive" than the solution we're providing so while I understand wanting to allow people to still use the flag here it will really do no good. But happy to put all of this block back on. It just seems very moot and possibly confusing to have it in there.

Reply:
Hey - we decided to prioritize the deletion of `--non-destructive` for the Wilt Chamberlain release since my previous comment here. Let's just remove all of the non-destructive code here :) 

See https://github.com/fishtown-analytics/dbt/issues/1415

Reply:
this is music to my ears!